Tasks/PowerShellOnTargetMachines2/task.json

```diff
@@ -0,0 +1,201 @@
+        }
+    ],
+    "releaseNotes": "What's new in Version 3.0: <br/>&nbsp;&nbsp; Use invoke-command to perform remote script execution",
+    "minimumAgentVersion": "2.0.0",
+    "inputs": [
+        {
+            "name": "Machines",
+            "type": "multiLine",
+            "label": "Machines",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Provide a comma separated list of machine IP addresses or FQDNs along with ports. Port is defaulted based on the selected protocol. <br>Eg: dbserver.fabrikam.com,dbserver_int.fabrikam.com:5986,192.168.12.34:5986 <br>Or provide output variable of other tasks. Eg: $(variableName) <br>If you are using HTTPS, name/IP of machine should match the CN in the certificate."
+        },
+        {
+            "name": "UserName",
+            "type": "string",
+            "label": "Admin Login",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Administrator login for the target machines. <br> Eg: Format: Domain\\Admin User, Admin User@Domain, .\\Admin User"
+        },
+        {
+            "name": "UserPassword",
+            "type": "string",
+            "label": "Password",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Administrator password for the target machines. <br>It can accept variable defined in Build/Release definitions as '$(passwordVariable)'. <br>You may mark variable type as 'secret' to secure it."
+        },
+        {
+            "name": "ScriptType",
+            "type": "radio",
+            "label": "Script Type",
+            "defaultValue": "FilePath",
+            "options": {
+                "FilePath": "File Path",
+                "Inline": "Inline"
+            },
+            "helpMarkDown": "The type of script to execute: Inline or File Path",
+            "groupName": "ScriptOptions"
+        },
+        {
+            "name": "ScriptPath",
+            "type": "string",
+            "label": "Script File Path",
+            "defaultValue": "",
+            "visibleRule": "ScriptType = FilePath",
+            "required": true,
+            "helpMarkDown": "Location of the PowerShell script on the target machines or on a UNC path like C:\\BudgetIT\\Web\\Deploy\\Website.ps1",
+            "groupName": "ScriptOptions"
+        },
+        {
+            "name": "InlineScript",
+            "type": "multiLine",
+            "label": "Script",
+            "visibleRule": "ScriptType = Inline",
+            "required": true,
+            "defaultValue": "# Write your powershell commands here.\n\nWrite-Output \"Hello World\"",
+            "properties": {
+                "resizable": "true",
+                "rows": "10",
+                "maxLength": "5000"
+            },
+            "helpMarkDown": "",
+            "groupName": "ScriptOptions"
+        },
+        {
+            "name": "ScriptArguments",
```

Comment:
should the visibility be there only for filePath?

Reply:
i know that the powershell task does not expose argument for inline script, but my personal opinion is that it comes in handy. And limiting its visibility to filepath will only cause people to maintain scripts separately for when they use vsts to run their scripts and when they are running it locally. I'll let @RoopeshNair  take a call on this.